buildinfo/buildinfo.go

```diff
@@ -65,16 +73,21 @@ func Version() string {
 func VersionsMatch(v1, v2 string) bool {
 	// Developer versions are disregarded...hopefully they know what they are
 	// doing.
-	if strings.HasPrefix(v1, develPrefix) || strings.HasPrefix(v2, develPrefix) {
+	if IsDevVersion(v1) || IsDevVersion(v2) {
```

Comment:
I think this logic made sense for `v0.0.0`, but for non-`v0.0.0` we know approximately the version in question and we could require the minor version match. Or is there a use-case for returning a match for e.g. `v1.1.1-devel` and `v1.2.3` as tested below?

Reply:
I was thinking along the same lines. I will change it so only v0.0.0 matches all, and other versions do check major minor as expected.

Reply:
This should be pretty much the previous behavior.